compiler/luci/pass/src/CircleOptimizer.cpp

```diff
@@ -158,6 +158,17 @@ void CircleOptimizer::quantize(loco::Graph *g) const
                                            str_to_granularity(granularity));
     quantizer.run(g);
   }
+
+  logo::Phase phase;
+
+  // Do Shape/Type inference (for validation)
```

Comment:
Don't exactly undersand why this is `for validation`.  May be add some more comments as notes?

Reply:
This is for passing `luci::validate`.

`circle-quantizer` will call `luci::validate` for validation after quantization (like `circle2circle` calls `luci::validate` after optimization). To pass the `luci::validate`, type and shape must be inferred in advance.

Reply:
Purpose of the shape inference and dtype inference is to set dtype and shape information to nodes that may not have it for new added by some transformations. 
Purpose of `validate` is to do minimal validation of the graph, including shape and dtype existance.
So doing shape and dtype inference actually is not for validation but for the model itself.
Here, dtype may be updated so may have to run dtype inference to update the information.
For shape, I'm not sure we need to run.

As for conclusion of the question, I think we can remove th comment `(for validation)`

Reply:
I think shape inference should be executed. If only quantize is called, export may require both inference...